base/shell/explorer/traywnd.cpp

```diff
@@ -365,7 +366,15 @@ class CTrayWindow :
 
         m_RunFileDlgOwner = hwnd;
 
-        RunFileDlg(hwnd, NULL, NULL, NULL, NULL, RFF_CALCDIRECTORY);
+        WCHAR szHomeDrive[MAX_PATH], szHomePath[MAX_PATH];
+        GetEnvironmentVariableW(L"HOMEDRIVE", szHomeDrive, _countof(szHomeDrive));
+        GetEnvironmentVariableW(L"HOMEPATH", szHomePath, _countof(szHomePath));
+
+        WCHAR szDefaultDir[MAX_PATH];
+        StringCchCopyW(szDefaultDir, _countof(szDefaultDir), szHomeDrive);
+        StringCchCatW(szDefaultDir, _countof(szDefaultDir), szHomePath);
+
+        RunFileDlg(hwnd, NULL, szDefaultDir, NULL, NULL, RFF_CALCDIRECTORY);
```

Comment:
Interesting observation:
- explorer.exe indeed uses:
```c
RunFileDlg(hWnd, NULL, "C:\Users\username", "Execute", NULL, RFF_CALCDIRECTORY)
```
- while taskmgr, from the "File - New Task" menu item, uses:
```c
RunFileDlg(hWnd, ptr1, "C:\Windows\system32", "Create a new task", "Dialog description", RFF_CALCDIRECTORY | RFF_NOSEPARATEMEM | 64)
```

Reply:
Shall we change "File - New Task" menu item action?